h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -814,16 +814,16 @@ private FVecParseWriter makeDout(ParseSetup localSetup, int chunkOff, int nchunk
           localSetup = ParserService.INSTANCE.getByInfo(localSetup._parse_type).setupLocal(vec,localSetup);
           // Zipped file; no parallel decompression;
           InputStream bvs = vec.openStream(_jobKey);
-          ZipInputStream zis = new ZipInputStream(bvs);
+          InputStream dec = decryptionTool.decryptInputStream(bvs);
```

Comment:
Just want to make sure if I understand it correctly: if encryption is disabled, `decryptionTool` is an instance of `NullDecryptionTool`?

Reply:
@mn-mikke yes, there is always an instance of the "decryption tool"